<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebuggerWindow</class>
 <widget class="QMainWindow" name="DebuggerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1377</width>
    <height>777</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>vm-tutorial debugger</string>
  </property>
  <widget class="QWidget" name="centralwidget"/>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPause_Continue"/>
   <addaction name="actionRun_to_Cursor"/>
   <addaction name="separator"/>
   <addaction name="actionStep_Into"/>
   <addaction name="actionStep_Out"/>
   <addaction name="actionStep_Over"/>
   <addaction name="separator"/>
   <addaction name="actionGo_to_PC"/>
   <addaction name="actionGo_to_Address"/>
   <addaction name="separator"/>
   <addaction name="actionTrace"/>
   <addaction name="actionConfigure"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_4">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Disassembly</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QTreeView" name="disasmView">
    <attribute name="headerVisible">
     <bool>false</bool>
    </attribute>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_5">
   <property name="maximumSize">
    <size>
     <width>220</width>
     <height>524287</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Registers</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QTreeView" name="registerView"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Memory</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="MemoryViewWidget" name="memoryView"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Breakpoints</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QTreeWidget" name="breakpointsWidget">
    <property name="columnCount">
     <number>2</number>
    </property>
    <column>
     <property name="text">
      <string>#</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Address</string>
     </property>
    </column>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_7">
   <property name="maximumSize">
    <size>
     <width>220</width>
     <height>524287</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Stack</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QTreeView" name="stackView"/>
  </widget>
  <action name="actionPause_Continue">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/pause_continue_button.png</normaloff>:/assets/pause_continue_button.png</iconset>
   </property>
   <property name="text">
    <string>Pause/Continue</string>
   </property>
   <property name="toolTip">
    <string>Pauses or continues the emulation from a debugged state.</string>
   </property>
  </action>
  <action name="actionStep_Into">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/step_into_button.png</normaloff>:/assets/step_into_button.png</iconset>
   </property>
   <property name="text">
    <string>Step Into</string>
   </property>
   <property name="toolTip">
    <string>Steps until the first line of a subroutine call.</string>
   </property>
  </action>
  <action name="actionStep_Over">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/step_over_button.png</normaloff>:/assets/step_over_button.png</iconset>
   </property>
   <property name="text">
    <string>Step Over</string>
   </property>
   <property name="toolTip">
    <string>Execute the next subroutine and stop.</string>
   </property>
  </action>
  <action name="actionStep_Out">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/step_out_button.png</normaloff>:/assets/step_out_button.png</iconset>
   </property>
   <property name="text">
    <string>Step Out</string>
   </property>
   <property name="iconText">
    <string>Step Out</string>
   </property>
   <property name="toolTip">
    <string>Execute the next subroutine until execution returns to the caller.</string>
   </property>
  </action>
  <action name="actionRun_to_Cursor">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/run_to_cursor_button.png</normaloff>:/assets/run_to_cursor_button.png</iconset>
   </property>
   <property name="text">
    <string>Run to Cursor</string>
   </property>
   <property name="toolTip">
    <string>Runs until the statement selected by the cursor is reached.</string>
   </property>
  </action>
  <action name="actionTrace">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/trace_button.png</normaloff>:/assets/trace_button.png</iconset>
   </property>
   <property name="text">
    <string>Trace</string>
   </property>
   <property name="toolTip">
    <string>Start tracing the execution to a file.</string>
   </property>
  </action>
  <action name="actionConfigure">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/settings_button.png</normaloff>:/assets/settings_button.png</iconset>
   </property>
   <property name="text">
    <string>Configure</string>
   </property>
   <property name="toolTip">
    <string>Configures the debugger.</string>
   </property>
  </action>
  <action name="actionGo_to_PC">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/go_to_pc_button.png</normaloff>:/assets/go_to_pc_button.png</iconset>
   </property>
   <property name="text">
    <string>Go to PC...</string>
   </property>
   <property name="toolTip">
    <string>Jumps to the current program counter.</string>
   </property>
  </action>
  <action name="actionGo_to_Address">
   <property name="icon">
    <iconset resource="../assets/assets.qrc">
     <normaloff>:/assets/go_to_address_button.png</normaloff>:/assets/go_to_address_button.png</iconset>
   </property>
   <property name="text">
    <string>Go to Address</string>
   </property>
   <property name="toolTip">
    <string>Jumps to the specified address.</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MemoryViewWidget</class>
   <extends>QWidget</extends>
   <header>memory_view.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../assets/assets.qrc"/>
 </resources>
 <connections/>
</ui>
